File: commons/src/main/java/ru/starksoft/commons/ApiLevelUtils.java
package ru.starksoft.commons;

import android.os.Build;

public final class ApiLevelUtils {

	private ApiLevelUtils() {
		throw new UnsupportedOperationException();
	}

	public static boolean isN() {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N;
	}

	public static boolean isQ() {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q;
	}

	public static boolean isO() {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O;
	}

	public static boolean isM() {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
	}

	public static boolean isP() {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.P;
	}
}

File: commons/src/main/java/ru/starksoft/commons/ApiLevelUtils.kt
package ru.starksoft.commons

import android.os.Build
import androidx.annotation.ChecksSdkIntAtLeast

object ApiLevelUtils {

    @get:ChecksSdkIntAtLeast(api = Build.VERSION_CODES.N)
    val isN: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N

    @get:ChecksSdkIntAtLeast(api = Build.VERSION_CODES.Q)
    val isQ: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q

    @get:ChecksSdkIntAtLeast(api = Build.VERSION_CODES.O)
    val isO: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O

    @get:ChecksSdkIntAtLeast(api = Build.VERSION_CODES.M)
    val isM: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M

    @get:ChecksSdkIntAtLeast(api = Build.VERSION_CODES.P)
    val isP: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.P

}
